### PR TITLE
Kotlin coroutines and generic tests

### PR DESCRIPTION
**Pull request by hadilq**
Thank you for submitting a pull request! But first:

 - [X] Can you back your code up with tests?

Couldn't find Kotlin coroutines's tests and also a generic class test, so added them. Hope it's useful.

### CODE REVIEW

File: tests/src/test/kotlin/test/MatchersTest.kt [hunk 1]
```diff
@@ -35,6 +36,24 @@ class MatchersTest : TestBase() {
...
+            verify(this).classClosed(any())
+        }
+    }
+
+    @Test
+    fun anyCoroutinesClosedClass() {
+        mock<Methods>().apply {
+            runTest {
+                coroutinesClosed(Closed())
+                //TODO verify(this).coroutinesClosed(any())
```
Comment (TimvdLippe) [inline tests/src/test/kotlin/test/MatchersTest.kt:52]: Does this mean this currently doesn't work as expected? Do you want to submit a fix as well for this?

**Review by TimvdLippe — APPROVED** [review]
Tests look good, thanks! 1 question about the TODO.

File: tests/src/test/kotlin/test/MatchersTest.kt (hunk 1 above)
Comment (TimvdLippe) [reply tests/src/test/kotlin/test/MatchersTest.kt:52]: Let's uncomment the code here and add a `@Disabled` to this test with an explanation. Then we can fix that in the next PR.

Comment (hadilq) [reply tests/src/test/kotlin/test/MatchersTest.kt]: interestingly, I removed the TODO and it passed the test!! I cannot remember what I have done before :D

**Review by TimvdLippe — APPROVED** [review]
Thanks!

**Pull request merged**